working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1885,14 +1900,27 @@ Where `<keyword>` is `var` or `final` is treated like so:
     }
     ```
 
-#### Pattern-if statement
+#### If-case statement
 
 1.  Evaluate the `expression` producing `v`.
 
-2.  Match the `pattern` against `v`.
+2.  Match the `pattern` in the `caseHead` against `v`.
+
+3.  If the match succeeds:
+
+    1.  If there is a guard clause:
+
+        1.  Evaluate it. If it does not evaluate to a Boolean, throw a runtime
+            exception. *This can happen if the guard expression's type is
+            `dynamic`.*
 
-3.  If the match succeeds, evaluate the then `statement`. Otherwise, if there
-    is an `else` clause, evaluate the else `statement`.
+        1.  If the guard evaluates to `true`, evaluate the then `statement`.
```

Comment:
Nit: evaluate -> execute (which is what we say we do for statements, evaluation is for expressions).

Reply:
I like that. Done.